### DISCUSSION TITLE
Error in the useContractWrite docs.

### DISCUSSION THREAD

**Discussion by equalogix** [wevm/wagmi#2211] [General]
There appears to be a small error on the `useContractWrite` page. In the example for the specific features like `args` and `overrides`. As shown here:
```
import { useContractWrite } from 'wagmi'
 
function App() {
  const contractWrite = useContractWrite({
    mode: 'recklesslyUnprepared',
    address: '0xecb504d39723b0be0e3a9aa33d646642d1051ee1',
    abi: wagmigotchiABI,
    functionName: 'feed',
  })
 
  return <button onClick={() => write()}>Feed</button>
}
```
The `usecontractWrite` hook is assigned to the const `contractWrite`, but in the button the `onClick` is calling `write`. It should either be `contractWrite.write()` or the write should be destructured like in the example at the top of the page. It's not a big deal, but could be confusing to beginners.

**Comment by vaulche-team** [comment]
This is work fine for my code
  const handleNFTSubmit = async (e: any) => {
    e.preventDefault();
    const MintInfo = [];
    MintInfo.push(formData);
    const dataToUpload = [MintInfo];
    try {
      const uris = await upload({ data: dataToUpload });
      console.log(uris, "...........");
      setIpfsdata(uris)
      return uris;
    } catch (error) {
      console.error("Upload Error:", error);
      return null;
    }
  };

  // MintNft function
  const Createitem = async (e: any) => {
    e.preventDefault();
    console.log(Ipfsdata,"this is the ipfsdata....")
    try {
      let value: any = await  handleNFTSubmit(e);
      console.log(value[0],"This is the data..")
      const { request } = await prepareWriteContract({
        address: NFT721,
        abi: Nftjson,
        functionName: "mintNft",
        args:[Ipfsdata[0]],
      });
      const { hash } = await writeContract(request);
      console.log(hash, "This is the transition data");
    } catch (error) {
      console.log(error, "Error occurred during creation");
    }
  };
